Document @edgio/sanity-studio connector for Sanity Apps (#644)
edgio-docs/edgio-docs: src/pages/guides/sites_frameworks/getting_started/sanity.md

This guide shows you how to deploy a [Sanity Studio](https://www.sanity.io/docs/sanity-studio) application to {{ PRODUCT }}.

Sanity Studio is a single page app (SPA) written in React, where you can configure the document types and input fields, with simple JavaScript objects. This guide will walk you through how to deploy Sanity Studio with {{ PRODUCT }}.

## Example {/*example*/}

<ExampleButtons
  title="Sanity Studio"
  siteUrl="https://layer0-docs-layer0-sanity-studio-example-default.layer0-limelight.link"
  repoUrl="https://github.com/edgio-docs/edgio-sanity-studio-example"
  deployFromRepo />

## Connector {/*connector*/}

> This framework has a connector developed for {{ PRODUCT }}. See [Connectors](connectors) for more information.

Reviewer: ```suggestion
This framework has a connector developed for {{ PRODUCT }}. See [Connectors](/guides/sites_frameworks/connectors) for more information.
```

<ButtonLink variant="stroke" type="code" withIcon={true} href="https://github.com/edgio-docs/edgio-connectors/tree/main/edgio-sanity-studio-connector">
 View the Connector Code
</ButtonLink>

{{ PREREQ }}

> ## Create a new Sanity Studio app {/*create-a-new-sanity-studio-app*/}

Reviewer: Is this article geared towards Sanity Studio v2 or v3? We should identify the version in the title. If we plan on supporting v3 instead of v2 which has been deprecated, then we should update this entire section to reflect the new npm workflow (as noted by their CLI).

> ⚠  Welcome to Sanity! Looks like you're following instructions for Sanity Studio v2,  ║
> ║    but the version you have installed is the latest, Sanity Studio v3.                ║
> ║    In Sanity Studio v3, new projects are created with [npm create sanity@latest].     ║
> ║                                                                                       ║
> ║    Learn more about Sanity Studio v3: https://www.sanity.io/help/studio-v2-vs-v3

Reviewer: The above comment applies to the section that starts on line 27.

Author: @jsernaque Ooo, interesting, thanks! I'll check this out.

- The `{{ PACKAGE_NAME }}/core` package
- The `{{ PACKAGE_NAME }}/cli` package
- The `{{ PACKAGE_NAME }}/sanity-studio` package
- `{{ CONFIG_FILE }}`- Contains various configuration options for {{ PRODUCT }}.
- `routes.js` - A default routes file that sends all requests to the Sanity Studio. Update this file to add caching or proxy some URLs to a different origin.

## Routing {/*routing*/}

The default `routes.js` file created by `{{ FULL_CLI_NAME }} init` sends all requests to Sanity Studio server via a fallback route.

```js
const { Router } = require('{{ PACKAGE_NAME }}/core/router')
const { sanityRoutes } = require('{{ PACKAGE_NAME }}/sanity-studio')

export default new Router({ indexPermalink: false }).use(sanityRoutes)
```

## Running Locally {/*running-locally*/}

To test your app locally, run:

```bash
{{ FULL_CLI_NAME }} run
```

You can do a production build of your app and test it locally using:

```bash
{{ FULL_CLI_NAME }} build && {{ FULL_CLI_NAME }} run --production
```

Setting `--production` runs your app exactly as it will be when deployed to the {{ PRODUCT }} cloud.

## Deploy to {{ PRODUCT }} {/*deploy-to*/}

> Deploy your app to the {{ PRODUCT_PLATFORM }} by running the following commands in your project's root directory:

Reviewer: ```suggestion
Deploy your app to the {{ PRODUCT }} cloud by running the following command in your project's root directory:
```

```bash
{{ FULL_CLI_NAME }} deploy
```

> See [deploying](deploy_apps) for more information.

Reviewer: ```suggestion
See [deploying](/guides/basics/deployments) for more information.
```
